src/DynamoCore/Graph/Nodes/NodeInputData.cs

```diff
@@ -134,7 +134,7 @@ public override bool Equals(object obj)
                 this.StepValue == converted.StepValue &&
                 this.Type == converted.Type &&
                 //check if the value is the same or if the value is a number check is it similar
-                ((this.Value == converted.Value) || valNumberComparison || this.Value.ToString() == this.Value.ToString());
+                ((this.Value == converted.Value) || valNumberComparison || this.Value.ToString() == converted.Value.ToString());
```

Comment:
whats up with this?

Reply:
seems like a bug fix!

Reply:
Ahhh yes.  This was a bug fix. that I found when Making the NodeOutputData class

Reply:
@ramramps @mjkkirschner This bug fix is causing a lot of serialization test failures.  Seems like I should reverse the fix and file another task to fix this equals override and all other bugs it exposes in serialization tests.

Reply:
Yeah, this part of the check was essentially a no-op before.  There was no comparison whatsoever on the `Value` field.